### PR TITLE
Programming Your Radio

### PR DESCRIPTION
**Pull request by jabbate19**

### CODE REVIEW

File: source/docs/networking/programming-your-radio.rst [hunk 1]
```diff
@@ -128,7 +122,6 @@ Use the Start menu or desktop shortcut to launch the program.
 
 | Note: If you need to locate the program it is installed to C:Files
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: You may want to make this into a "note" block with 
```
.. note:: this is a note block
 it contains notes.
```

File: source/docs/networking/programming-your-radio.rst [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+**The OM5P-AN and AC use the same power plug as the D-Link DAP1522,
+however they are 12V radios. Wire the radio to the 12V 2A terminals on
+the VRM (center-pin positive).**
+
+**Note: Teams will need to update firmware on both OM5P-AN and OM5P-AC
+radios in order for the programming utility to program them, or for them
+to be used at events. This must be done before you attempt to program
+them.**
+Pre-Requisites
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need a blank line above titles

File: source/docs/networking/programming-your-radio.rst [hunk 3]
```diff
@@ -0,0 +1,304 @@
+Programming Your Radio
+======================
+
+This guide will show you how to use the FRC Radio Configuration Utility
+software to configure your robot's wireless bridge for use outside of
+FRC events.
+
+Before you begin using the software:
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: This should be a subtitle

File: source/docs/networking/programming-your-radio.rst [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+Disable WiFi connections on your computer, as it may prevent the
+configuration utility from properly communicating with the bridge Make
+sure no devices are connected to your computer via ethernet, other than
+the wireless bridge. \*\*Note that for the OM5P-AN and AC bridge, it is
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Replace with
```
.. note:: Note that for the OM5P-AN and AC bridge, it is
 recommended to use a particular Ethernet port. See the on screen image
 and instructions for more information.
```

File: source/docs/networking/programming-your-radio.rst (hunk 2 above)
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: This should be a subtitle, also needs spacing above it.

File: source/docs/networking/programming-your-radio.rst [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+The FRC Radio Configuration Utility requires the Java Runtime Engine
+(JRE). If you do not have Java installed, you can download the JRE from
+here: https://www.java.com/en/download/
+
+The FRC Radio Configuration Utility requires Administrator privileges to
+configure the network settings on your machine. The program should
+request the necessary privileges automatically (may require a password
+if run from a non-Administrator account), but if you are having trouble,
+try running it from an Administrator account.
+Application Notes
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: This should be a subtitle

File: source/docs/networking/programming-your-radio.rst [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+By default, the Radio Configuration Utility will program the radio to
+enforce the 4Mbps bandwidth limit on traffic exiting the radio over the
+wireless interface. In the home configuration (AP mode) this is a total,
+not a per client limit. This means that streaming video to multiple
+clients is not recommended.
+
+The Utility has been tested on Windows 7, 8 and 10. It may work on other
+operating systems, but has not been tested.
+Programmed Configuration
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need line break/space above it

File: source/docs/networking/programming-your-radio.rst [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+   as part of the DHCP.
+
+At home only:
+
+-  SSID may have a "Robot Name" appended to the team number to
+   distinguish multiple networks.
+-  Firewall option may be enabled to mimic the field firewall rules
+   (open ports may be found in the Game Manual)
+
+**Note: It is not possible to modify the configuration manually**
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Replace with 
`.. note:: Note: It is not possible to modify the configuration manually`

File: source/docs/networking/programming-your-radio.rst [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+Download the latest FRC Radio Configuration Utility Installer from the
+following links:
+
+`FRC Radio Configuration
+19.1.1 <http://https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1.zip>`__
+
+`FRC Radio Configuration 19.1.1 Israel
+Version <http://https://firstfrc.blob.core.windows.net/frc2019/Radio/FRC_Radio_Configuration_19_1_1_IL.zip>`__
+
+\*\*Note: The \_IL version is for Israel teams and contains a version of
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Replace with a note block

File: source/docs/networking/programming-your-radio.rst [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+the OM5PAC firmware with restricted channels for use in Israel.\*\*
+
+Version 19.1.1 corrects an issue with applying the Bandwidth Limit
+present in version 19.1.0. Teams should install the new version, then
+re-program their radio (their is no need to re-flash the firmware).
+Install the Software
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need a line break/space above it

File: source/docs/networking/programming-your-radio.rst [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/pyr3.png
+   :width: 600
+   
+Use the Start menu or desktop shortcut to launch the program.
+
+| Note: If you need to locate the program it is installed to C:Files
+(x86)Radio Configuration Utility. For 32-bit machines the path is C:FilesRadio Configuration Utility\
+Allow the Program to Make Changes, If Prompted
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/line break above it

File: source/docs/networking/programming-your-radio.rst [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/pyr4.png
+   :width: 600
+   
+If the your computer is running Windows Vista or Windows 7, a prompt may appear
+about allowing the configuration utility to make changes to the
+computer. Click "Yes" if the prompt appears.  
+Select the Network Interface
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/line break above it.

File: source/docs/networking/programming-your-radio.rst [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+the configuration utility will use to communicate with the wireless
+bridge. On Windows machines, ethernet interfaces are typically named
+"Local Area Connection". The configuration utility can not program a
+bridge over a wireless connection.
+
+1. If no ethernet interfaces are listed, click "Refresh" to re-scan for
+   available interfaces
+2. Select the interface you want to use from the drop-down list
+3. Click "OK"
+Open Mesh Firmware Note
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 13]
```diff
@@ -0,0 +1,304 @@
...
+OpenWRT firmware. **OM5P-AC radios in the 2019 KoP should not need an
+update.**
+
+If you do not need to update or re-load the firmware, skip the next
+step.
+
+Note: Radios used in 2018 will need to be updated to 2019 firmware
+before configuring.
+Loading FRC Firmware to OpenMesh Radio
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 14]
```diff
@@ -0,0 +1,304 @@
...
+than the one being used to program the radio. If only one adapter is
+found, the tool should attempt to use that one. See the steps at the
+bottom of this article for more info.
+
+Teams may also see this error with foreign language Operating Systems.
+If you experience issues loading firmware or programming on a foreign
+language OS, try using an English OS, such as on the KOP provided PC or
+setting the Locale setting to "en\_us" as described on `this
+page <http://https://www.java.com/en/download/help/locale.xml>`__.
+Select a Bridge Model and Operating Mode
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 15]
```diff
@@ -0,0 +1,304 @@
...
+
+.. image:: images/pyr7.png
+   :width: 600
+   
+1. Select which radio you are configuring using the drop-down list.
+2. Select which operating mode you want to configure. For most cases,
+   the default selection of 2.4GHz Access Point will be sufficient. If
+   your computers support it, the 5GHz AP mode is recommended, as 5GHz
+   is less congested in many environments.
+Select Options
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 16]
```diff
@@ -0,0 +1,304 @@
...
+2. Firewall: If this box is checked, the radio firewall will be
+   configured to attempt to mimic the port blocking behavior of the
+   firewall present on the FRC field. For a list of open ports, please
+   see the FRC Game Manual.
+3. BW Limit: If this box is checked, the radio enforces a 4MB/s
+   bandwidth limit like it does when programmed at events. Note that in
+   AP mode, this is a total limit, not per client, so streaming video to
+   multiple clients simultaneously may cause undesired behavior.
+
+**Note: Firewall and BW Limit only apply to the OpenMesh radios. These
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Replace with note block

File: source/docs/networking/programming-your-radio.rst [hunk 17]
```diff
@@ -0,0 +1,304 @@
...
+options have no effect on D-Link radios**
+
+The "Firewall" option configures the radio to emulate the field
+firewall. This means that you will not be able to deploy code wirelessly
+with this option enabled.
+Prepare and Start the Configuration Process
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 18]
```diff
@@ -0,0 +1,304 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/pyr9.png
+   :width: 600
+   
+Follow the on-screen instructions for preparing your wireless bridge,
+entering the settings the bridge will be configured with, and starting
+the configuration process. These on-screen instructions update to match
+the bridge model and operating mode chosen.
+Configuration Progress
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 19]
```diff
@@ -0,0 +1,304 @@
...
+
+.. image:: images/pyr10.png
+   :width: 600
+   
+Throughout the configuration process, the window will indicate:
+
+1. The step currently being executed
+2. The overall progress of the configuration process
+3. All steps executed so far
+Configuration Completed
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 20]
```diff
@@ -0,0 +1,304 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/pyr11.png
+   :width: 600
+   
+Once the configuration is complete:
+
+1. Press "OK" on the dialog window
+2. Press "OK" on the main window to return to the settings screen
+Configuration Errors
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Need space/linebreak above it

File: source/docs/networking/programming-your-radio.rst [hunk 21]
```diff
@@ -0,0 +1,304 @@
...
+^^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/pyr12.png
+   :width: 600
+   
+If an error occurs during the configuration process, follow the
+instructions in the error message to correct the problem.
+
+ Troubleshooting: Disabling Network Adapters
```
Comment (Daltz333) [inline source/docs/networking/programming-your-radio.rst]: Should be a subtitle

**Comment by Daltz333** [comment]
Also, please target the *develop* branch instead of master. Thanks 😄

**Comment by mcm001** [comment]
If this is abandoned I'm happy to pick it up

**Comment by Daltz333** [comment]
@jabbate19 Update before I mark this as stale?

**Comment by jabbate19** [comment]
So sorry for no response, I’ve been out and about for so long I haven’t been touching code. If you want it you can take it, I’m stacked on APs rn

**Comment by mcm001** [comment]
I've gone ahead and fixed all the style stuff and converted the note blocks. I've put it in my fork of frc-docs in a radio branch.

**Comment by mcm001** [comment]
See #28

**Pull request closed without merging**